GVFS/GVFS/CommandLine/SparseVerb.cs

```diff
@@ -235,10 +237,105 @@ protected override void Execute(GVFSEnlistment enlistment)
                     {
                         this.PruneFoldersOutsideSparse(tracer, enlistment, sparseTable);
                     }
+
+                    if (needToChangeProjection || this.Prune)
+                    {
+                        // Update the last write times of the parents of folders being added/removed
+                        // so that File Explorer will refresh them
+                        UpdateParentFolderLastWriteTimes(tracer, enlistment.WorkingDirectoryBackingRoot, foldersToRemove, foldersToAdd);
+                    }
                 }
             }
         }
 
+        private static void UpdateParentFolderLastWriteTimes(ITracer tracer, string rootPath, List<string> foldersToRemove, List<string> foldersToAdd)
+        {
+            Stopwatch updateTime = Stopwatch.StartNew();
+
+            HashSet<string> foldersToRefresh = new HashSet<string>(GVFSPlatform.Instance.Constants.PathComparer);
+            foreach (string path in foldersToRemove)

```

Comment:
I would either
```
foreach (string path in foldersToRemove.Concat(foldersToAdd).Concat(foldersPruned))
```
or make `AddNonRootParentPathsToSet` take the `List<string>` and have it do the `foreach` so that you would have
```
AddNonRootParentPathsToSet(foldersToRemove, foldersToRefresh);
AddNonRootParentPathsToSet(foldersToAdd, foldersToRefresh);
AddNonRootParentPathsToSet(foldersPruned, foldersToRefresh);
```